### ISSUE TITLE
stuck at connecting again.

### ISSUE THREAD

**Issue by Ninjastorm380** [yakyak/yakyak#1013]
not even 2 weeks and yakyak is stuck at the connecting screen. i have already applied the DS key fix, did google do it again?
someone please help!

**Comment by kokkinosb** [comment]
Same here

**Comment by kaldoran** [comment]
Same here, looks like google did change something again.

Here is the error : 

> connect_failed TypeError: Cannot read property 'length' of null
    at /home/kaldoran/yakyak/app/node_modules/hangupsjs/lib/init.js:125:21
    at _fulfilled (/home/kaldoran/yakyak/app/node_modules/q/q.js:854:54)
    at /home/kaldoran/yakyak/app/node_modules/q/q.js:883:30
    at Promise.promise.promiseDispatch (/home/kaldoran/yakyak/app/node_modules/q/q.js:816:13)
    at /home/kaldoran/yakyak/app/node_modules/q/q.js:570:49
    at runSingle (/home/kaldoran/yakyak/app/node_modules/q/q.js:137:13)
    at flush (/home/kaldoran/yakyak/app/node_modules/q/q.js:125:13)
    at process._tickCallback (internal/process/next_tick.js:61:11)
error connecting TypeError: Cannot read property 'length' of null
    at /home/kaldoran/yakyak/app/node_modules/hangupsjs/lib/init.js:125:21
    at _fulfilled (/home/kaldoran/yakyak/app/node_modules/q/q.js:854:54)
    at /home/kaldoran/yakyak/app/node_modules/q/q.js:883:30
    at Promise.promise.promiseDispatch (/home/kaldoran/yakyak/app/node_modules/q/q.js:816:13)
    at /home/kaldoran/yakyak/app/node_modules/q/q.js:570:49
    at runSingle (/home/kaldoran/yakyak/app/node_modules/q/q.js:137:13)
    at flush (/home/kaldoran/yakyak/app/node_modules/q/q.js:125:13)
    at process._tickCallback (internal/process/next_tick.js:61:11)

**Comment by kaldoran** [comment]
I did fix the problem. This PR fixed it 
https://github.com/yakyak/hangupsjs/pull/107

I did remove the 'node_modules' files, from app and base directory.
`rm -rf node_modules `

And then reinstall yakyak 'npm i'

**Comment by hell-spawn** [comment]
I have same problem to @kaldoran , 

> `connect_failed TypeError: Cannot read property 'length' of null`
`    at /home/mauricio.ariza/Apps/yakyak-linux-x64/resources/app/node_modules/hangupsjs/lib/init.js:135:21`
`    at _fulfilled (/home/mauricio.ariza/Apps/yakyak-linux-x64/resources/app/node_modules/hangupsjs/node_modules/q/q.js:854:54)`
`    at /home/mauricio.ariza/Apps/yakyak-linux-x64/resources/app/node_modules/hangupsjs/node_modules/q/q.js:883:30`
`    at Promise.promise.promiseDispatch (/home/mauricio.ariza/Apps/yakyak-linux-x64/resources/app/node_modules/hangupsjs/node_modules/q/q.js:816:13)`
`   at /home/mauricio.ariza/Apps/yakyak-linux-x64/resources/app/node_modules/hangupsjs/node_modules/q/q.js:570:49`
`   at runSingle (/home/mauricio.ariza/Apps/yakyak-linux-x64/resources/app/node_modules/hangupsjs/node_modules/q/q.js:137:13)`
`   at flush (/home/mauricio.ariza/Apps/yakyak-linux-x64/resources/app/node_modules/hangupsjs/node_modules/q/q.js:125:13)`
`   at process._tickCallback (internal/process/next_tick.js:61:11)`

105 self_entity: { 
106               key: 'ds:19',
107               fn: function(d) {
108                 return CLIENT_GET_SELF_INFO_RESPONSE.parse(d[0]).self_entity;
109               }

**"d"  value is  [ [ 'cin:opc', '' ] ]** 

what have to go here?

**Comment by cout** [comment]
The PR in  yakyak/hangupsjs#107 did fix it for me.  I don't really understand the code, but it looks like an improvement (names instead of magic numbers).

**Comment by hell-spawn** [comment]
> The PR in [yakyak/hangupsjs#107](https://github.com/yakyak/hangupsjs/pull/107) did fix it for me. I don't really understand the code, but it looks like an improvement (names instead of magic numbers).

Sorry I'm confused with the patch 104, and update and everything is ok.

**Comment by sooncj** [comment]
for those of you who need a patch

resources/app/node_modules/hangupsjs/lib/init.js
```
@@ -70 +70 @@
-              key: 'ds:7',
+              name: 'cin:cac',
@@ -76 +76 @@
-              key: 'ds:31',
+              name: 'cic:vd',
@@ -82 +82 @@
-              key: 'ds:2',
+              name: 'cin:acc',
@@ -88 +88 @@
-              key: 'ds:2',
+              name: 'cin:acc',
@@ -94 +94 @@
-              key: 'ds:4',
+              name: 'cin:bcsc',
@@ -100 +100 @@
-              key: 'ds:20',
+              name: 'cgsirp',
@@ -106 +106 @@
-              key: 'ds:19',
+              name: 'cgsirp',
@@ -112 +112 @@
-              key: 'ds:19',
+              name: 'cgsirp',
@@ -121 +121,6 @@
-              return spec.key === e.key;
+              if (spec.name && e.data()) {
+                d = e.data();
+                if (d.length > 0 && d[0].length > 0 && d[0][0].length > 0)
+                  return spec.name == d[0][0]
+              }
+              return spec.key == e.key;
```

**Comment by averissimo** [comment]
Are you guys using v1.5.3 or 1.5.4-beta-rolling?

The connection issues should be fixed since v1.5.3 *(I just tested and it works in my computer)*

**Comment by hammerandtongs** [comment]
@averissimo I think I was on a patched 1.5.2 (patching it with the above from @sooncj made it work) but testing 1.5.3 (which is what I should have tried first :\ ) today it works well for me.
